deis/README.md

```diff
@@ -0,0 +1,73 @@
+# Deploy a Deis cluster
+
+<a href="https://azuredeploy.net/" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+
+This template allows you to create a Deis cluster. This template also deploys a Storage Account, Virtual Network, Public IP addresses and a Network Interface. 
+
+##Deploy the cluster
+
+1. Generate an OpenSSH key pair:
+		ssh-keygen -t rsa -b 4096 -c "[your_email@domain.com]"
+2. Generate a certificate using the private key above:
+		openssl req -x509 -days 365 -new -key [your private key file] -out [cert file to be generated]
+3. Go to https://discovery.etcd.io/new to generate a new cluster token.
+4. Modify **cloud-config.yaml** to replace the existing **discovery** token with the new token.
+5. Modify **azuredeploy-parameters.json**: Open the certificate you created in step 2. Copy all text between  *----BEGIN CERTIFICATE-----* and *-----END CERTIFICATE-----* into the **sshKeyData** parameter (you'll need to remove all newline characters).
+6. Modify other parameters such as **newStorageAccountName**, **vmNamePrefix**, **virtualNetworkName** to values of your choice. 
+5. Provision the resource group:
+
+>Note: If you chose to use the "Deploy to Azure" button experience, you'll need to manually encode **cloud-config.yaml** as a Base64 string and enter the encoded string to the **customData** parameter. Although the template can be updated to use the built-in base64() founction, I found the triple-encoding is rather troublesome especially for readability and maintenance.
+		
+##Install the client
+You need **deisctl** to control your Deis cluster. *deisctl* is automatically installed in all the cluster nodes. However, it's a good practice to use *deisctl* on a separate administrative machine. Because all nodes are configured with public IP addresses, you'll be able to use *deisctl* from any client machines. The following are the steps of setting up *deisctl* on a separate machine.
```

Comment:
you should also be able to deploy a cluster without having public ips on each machine... last time I checked - azure only gives you 5 public IPs per account. Today if someone used the current scripts without the `-pip` command they would get a legacy cloud service which comes with a load balanced public ip.